src/main/java/com/monumental/services/MonumentService.java

```diff
@@ -535,7 +540,16 @@ public MonumentBulkValidationResult validateMonumentCSV(List<String[]> csvList,
 
         List<CsvMonumentConverterResult> results = CsvMonumentConverter.convertCsvRows(csvList, mapping, zipFile);
         for (int i = 0; i < results.size(); i++) {
-            monumentBulkValidationResult.getResults().put(i + 1, results.get(i));
+            CsvMonumentConverterResult result = results.get(i);
+
+            List<Monument> duplicates = this.findDuplicateMonuments(result.getMonument().getTitle(),
+                    result.getMonument().getLat(), result.getMonument().getLon(), result.getMonument().getAddress());
+
+            if (duplicates.size() > 0) {
+                result.getWarnings().add("Potential duplicate records detected for this row.");
```

Comment:
I think you should include links to the duplicates. I think you can use relative urls like `/monuments/1` but you might want to use something similar to `publicUrl` in the `UserService` so that links work while using the react dev server as well